### PR TITLE
Upgrade stories to CSF3

### PR DESCRIPTION
**Pull request by alex-yau-ttd**
- Run Storybook's migration to [Component Story Format 3](https://storybook.js.org/blog/storybook-csf3-is-here/), i.e. `npx storybook@latest migrate csf-2-to-3 --glob="**/*.stories.tsx" --parser=tsx`
- Make some adjustments to remove unused imports
- Follow existing pattern of strongly typed Storybook Meta, rather than exporting default with `as  Meta<typeof MyType>`
- Use `StoryObj` instead of `StoryFn` for simple cases
- Simplify some stories by using the spread operator for arguments.

### CODE REVIEW

File: src/web/components/SharingPermission/SharingPermissionsTable.stories.tsx [hunk 1]
```diff
@@ -59,15 +54,26 @@ const response: SharingSiteDTO[] = [
...
-  </TestAllSitesListProvider>
-);
 
-export const SharedWithParticipants = Template.bind({});
-SharedWithParticipants.args = {
-  sharedSiteIds: [10, 11, 12, 14, 15, 17],
-  sharedTypes: ['DSP', 'DATA_PROVIDER'],
-  onDeleteSharingPermission: () => Promise.resolve(),
+const meta: Meta<typeof SharingPermissionsTable> = {
+  title: 'Sharing Permissions/Sharing Permissions Table',
```
Comment (alex-yau-ttd) [inline src/web/components/SharingPermission/SharingPermissionsTable.stories.tsx:59]: Changed the title slightly

File: src/web/components/Account/CreateAccountForm.stories.tsx [hunk 2]
```diff
@@ -1,14 +1,15 @@
-import { ComponentMeta, ComponentStory } from '@storybook/react';
+import { Meta, StoryFn } from '@storybook/react';
 import { useState } from 'react';
 
 import { CreateAccountForm, CreateAccountFormProps } from './CreateAccountForm';
 
-export default {
+const meta: Meta<typeof CreateAccountForm> = {
```
Comment (alex-yau-ttd) [inline src/web/components/Account/CreateAccountForm.stories.tsx:6]: I do this type of change to pretty much every file in this PR FYI!

File: src/web/components/ClientSideTokenGeneration/CstgDomainsTable.stories.tsx [hunk 3]
```diff
@@ -25,9 +25,7 @@ export const WithoutDomains: Story = {
     domains: [],
     onUpdateDomains: (domains: string[]) => {
       Promise.resolve(console.log('update domains to:', domains));
```
Comment (ssundahlTTD) [inline src/web/components/ClientSideTokenGeneration/CstgDomainsTable.stories.tsx:27]: Not your code, but why is this console.log() in a Promise.resolve()?  Seems weird.

**Review by ssundahlTTD — APPROVED** [review]

File: src/web/components/ClientSideTokenGeneration/CstgDomainsTable.stories.tsx (hunk 3 above)
Comment (alex-yau-ttd) [reply src/web/components/ClientSideTokenGeneration/CstgDomainsTable.stories.tsx:27]: I think this is just our pattern to console log inputs in storybooks where the prop returns a promise. Is there a better way to do that?

**Pull request merged**